devlib/utils/misc.py

```diff
@@ -590,6 +591,31 @@ def __str__(self):
         return message.format(self.filepath, self.lineno, self.message)
 
 
+def load_struct_from_yaml(filepath):
+    '''
+    Parses a config structure from a YAML file.
+    The structure should be composed of basic Python types.
+
+    Args:
+        filepath (str): Input file which contains YAML data.
+
+    Raises:
+        LoadSyntaxError: if there is a syntax error in YAML data.
+
+    Returns:
+        dict: which contains parsed YAML data.
+    '''
+
+    try:
+        yaml = YAML(typ='safe', pure=True)
+        with open(filepath, 'r', encoding='utf-8') as file_handler:
+            return yaml.load(file_handler)
+    except yaml.YAMLError as ex:
+        message = ex.message if hasattr(ex, 'message') else ''
+        lineno = ex.problem_mark.line if hasattr(ex, 'problem_mark') else None
+        raise LoadSyntaxError(message, filepath=filepath, lineno=lineno) from ex
```

Comment:
I'm not sure what LoadSyntaxError was initially for. Currently it seems like dead code in devlib. Unless it really adds something, I'd recommend just letting the YAMLError bubble up.

Reply:
`LoadSyntaxError()` is referenced in `load_struct_from_python()`, too.
I don't know the history, but apparently it makes the syntax error location more explicit, IMHO.

Reply:
We use this in WA to try and provide a more friendly error message to clearly indicate the location of the syntax error. 
Since this code is being moved, I'd vote for keeping the same error class.

Reply:
Sounds good, I did not look beyond devlib codebase